scripts/runInsider.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+
+# Check for install/updates at https://github.com/insidersec/insider
+
+benchmark_version=$(scripts/getBenchmarkVersion.sh)
+insider_version=$(insider -version | grep Version | cut -d' ' -f2)
+result_file="results/Benchmark_$benchmark_version-insider-v$insider_version.json"
+
+insider -quiet --tech java -no-html -exclude '.idea' -exclude '.mvn' -exclude 'results' -exclude 'scorecard' -exclude 'scripts' -exclude 'tools' -target "."
+mv report.json "$result_file"
```

Comment:
Can you enhance this script to verify that insider is even installed, and if it is not installed, either scripts up the installation and asks you if you want to install it, or at least provides basic install instructions, and then exits.

Also, for this and Horusec (and others), if there is an easy way to update the tool, it's nice if the script tries to auto update the tool version periodically.

As an example, https://github.com/OWASP/Benchmark/blob/master/tools/Contrast/runBenchmark_wContrast.sh checks to see how old contrast.jar is and if more than 24 hours old, pulls a new version.

Reply:
Added check and notification if new version is available.

PS: Insider has Docker images, but they report themself as version "dev".

Reply:
I ran your updated script and got this:
scripts/runInsider.sh 
 
Insider not found. Visit https://github.com/insidersec/insider and follow installation instructions
scripts/runInsider.sh: line 8: insider: command not found
You're using insider , but 3.0.0 is available. You should update your installation.
scripts/runInsider.sh: line 16: insider: command not found
mv: rename report.json to results/Benchmark_1.2-insider-v.json: No such file or directory

Can you make sure that the script properly halts if 'insider' isn't found?  When I run: which insider.     I get:

insider not found

Reply:
should be fixed now. just learned that you shouldn't use which to check whether a program is installed 🙈

Reply:
After installing Insider, this worked fine, with results in the /results folder as expected.